### PR TITLE
Wip/hughsie/xmlb quirks

### PR DESCRIPTION
**Pull request by hughsie**
This fails in CI, but I wanted to sanity check. I've released libxmlb 0.1.13 this morning but it needs a few weeks/months to get into distros I guess.

### CODE REVIEW

**Comment by superm1** [comment]
Just got it into Debian at least. https://tracker.debian.org/news/1072326/accepted-libxmlb-0113-1-source-into-unstable/

**Comment by superm1** [comment]
I flagged arch out of date too, so that should hopefully get updated soon.  Can you get fedora sorted?

**Comment by hughsie** [comment]
> Can you get fedora sorted?

Updates already filed ;)

File: src/fu-self-test.c [hunk 1]
```diff
@@ -2141,29 +2139,24 @@ fu_plugin_quirks_func (void)
...
+	ret = fu_quirks_load (quirks, FU_QUIRKS_LOAD_FLAG_NONE, &error);
+	g_assert_no_error (error);
+	g_assert (ret);
 
 	/* lookup */
 	g_timer_reset (timer);
 	for (guint j = 0; j < 1000; j++) {
 		g_autofree gchar *group = NULL;
-		group = g_strdup_printf ("DeviceInstanceId=USB\\VID_0BDA&PID_%04X", j);
+		group = g_strdup_printf ("DeviceInstanceId=USB\\VID_0BDA&PID_1100");
```
Comment (superm1) [inline src/fu-self-test.c]: this probably could be a const string now.

**Comment by superm1** [comment]
@hughsie looks like all the CI targets now have a new enough libxmlb.  I checked this on the system I did emmc on and it does enumerate properly still

**Comment by hughsie** [comment]
> I checked this on the system I did emmc on and it does enumerate properly still

Right. At the moment in this branch we can only have one plugin per physical device. Maybe that makes sense, but the problem is that the "last set quirk match" wins -- it's not appended to plugin_name like it was before. So, in this example, we call `fu_device_quirks_iter_cb()` multiple times, and once with `Plugin=ata` and once with `Plugin=emmc` I think as they both match the class GUID.

I don't think it makes sense to have two FuDevices representing the same physical thing, although if there were two plugins like ata and emmc they wouldn't always share the same `FirmwareSizeMax` for instance. On the flip side, having one `DeviceId` that's not unique assigned to a plugin is problematic.

Perhaps the pragmatic thing is to allow a string list of plugin_names, but only allow a single `FuPlugin` to be responsible for the device.

**Comment by superm1** [comment]
> Perhaps the pragmatic thing is to allow a string list of plugin_names, but only allow a single FuPlugin to be responsible for the device.

The other alternative is to have a new storage plugin and make emmc and ata both derive from it.

**Comment by hughsie** [comment]
> The other alternative is to have a new storage plugin and make emmc and ata both derive from it.

I think the current flat model is as complicated as I'm comfortable with, if I'm honest.

**Comment by hughsie** [comment]
I'm basically happy with this now.

**Comment by superm1** [comment]
LGTM thanks.

**Pull request merged**